lib/tree.go

```diff
@@ -96,6 +96,15 @@ func (t *Tree) checkUnresolved() {
 // add adds a leaf to the tree
 func (t *Tree) add(leaf *Leaf) *Tree {
 	t.checkName(leaf.name)
-	t.leaves = append(t.leaves, leaf)
+
+	ptr := leaf.structurePointer
+
+	if _, ok := t.leaves[ptr]; ok {
+		t.leaves[ptr].AddAlias(leaf.name)
+		return t
+	}
+
+	t.leaves[ptr] = leaf
```

Comment:
I found this line a bit confusing to me. We're using a pointer as the key for a map and the real structure as value? 
I don't think we are using this pointer anywhere, so why not use the leaf name as key?

Reply:
The block before this line is using it to check if it has already been found. I now realize that it will be easier to read if I call the variable address, since it isn't actually a pointer.